bmt/bmt.go

```diff
@@ -434,33 +483,84 @@ type SectionWriter interface {
 // * it will not leak processes if not all sections are written but it blocks
 //   and keeps the resource which can be released calling Reset()
 type AsyncHasher struct {
-	*Hasher            // extends the Hasher
-	mtx     sync.Mutex // to lock the cursor access
-	double  bool       // whether to use double segments (call Hasher.writeSection)
-	secsize int        // size of base section (size of hash or double)
-	write   func(i int, section []byte, final bool)
+	*Hasher             // extends the Hasher
+	mtx      sync.Mutex // to lock the cursor access
+	double   bool       // whether to use double segments (call Hasher.writeSection)
+	secsize  int        // size of base section (size of hash or double)
+	seccount int        // base section count
+	write    func(i int, section []byte, final bool)
+	errFunc  func(error)
+	all      bool // if all written in one go, temporary workaround
+	jobSize  int
+}
+
+// Reset implements param.SectionWriter
+func (sw *AsyncHasher) Reset() {
+	sw.jobSize = 0
+	sw.all = false
+	sw.Hasher.Reset()
 }
 
-// methods needed to implement AsyncWriter
+// SetLength implements param.SectionWriter
+func (sw *AsyncHasher) SetLength(length int) {
+	sw.jobSize = length
+}
+
+// SetWriter implements param.SectionWriter
+func (sw *AsyncHasher) SetWriter(_ param.SectionWriterFunc) param.SectionWriter {
+	sw.errFunc(errors.New("Asynchasher does not currently support SectionWriter chaining"))
```

Comment:
Error should not start with a capital letter.

Reply:
You mean the error string?

Reply:
You mean the error string?

Reply:
Yes, the error string.